src/components/application_manager/include/application_manager/application.h

```diff
@@ -366,12 +366,6 @@ class DynamicApplicationData {
    */
   virtual smart_objects::SmartObject FindSubMenu(uint32_t menu_id) const = 0;
 
-  /*
-   * @brief Returns true if sub menu with such name already exist
-   */
-  virtual bool IsSubMenuNameAlreadyExist(const std::string& name,
-                                         const uint32_t parent_id) = 0;
```

Comment:
Unfortunately, this function is public, so we need to deprecate it rather than remove it

Reply:
Do we need to do the same with the CoincidencePredicates in create_interaction_choice_set_request.h

Reply:
@iCollin nope, those are private